src/client.zig

```diff
@@ -120,11 +131,19 @@ pub const Client = struct {
         user_data: u128,
         callback: Request.Callback,
         operation: Operation,
-        data: []const u8,
+        message: *Message,
+        body_length: usize,
     ) void {
-        const message = self.message_bus.get_message() orelse
-            @panic("TODO: bubble up an error/drop the request");
-        defer self.message_bus.unref(message);
+        const message_size = @intCast(u32, @sizeOf(Header) + body_length);
+        if (message_size > config.message_size_max) {
+            callback(
+                user_data,
+                operation,
+                empty_result[0..],
+                @enumToInt(ClientError.max_message_size_exceeded),
+            );
+            return;
+        }
```

Comment:
Great to see this check here.

Let's just convert this into an assert, since it should already be impossible for the user to overflow the `message.body()` slice when they work with it. If `@sizeOf(Header) + body_length` overflows `config.message_size_max` then it means they miscalculated `body_length` or passed the wrong variable to this function. So this is programming error as opposed to operational error, and a panic via assertion would be better.